src/Orchard.Web/Modules/Orchard.MediaLibrary/Security/MediaAuthorizationEventHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Orchard.ContentManagement;
+using Orchard.MediaLibrary.Models;
+using Orchard.MediaLibrary.Services;
+using Orchard.Security;
+
+namespace Orchard.MediaLibrary.Security {
+    public class MediaAuthorizationEventHandler : IAuthorizationServiceEventHandler {
+        private readonly IAuthorizer _authorizer;
+        private readonly IMediaLibraryService _mediaLibraryService;
+
+        public MediaAuthorizationEventHandler(
+            IAuthorizer authorizer,
+            IMediaLibraryService mediaLibraryService) {
+            _authorizer = authorizer;
+            _mediaLibraryService = mediaLibraryService;
+        }
+
+        public void Checking(CheckAccessContext context) { }
+        public void Complete(CheckAccessContext context) { }
+
+        public void Adjust(CheckAccessContext context) {
+            if (context.Content.Is<MediaPart>()) {
+                var mediaPart = context.Content.As<MediaPart>();
```

Comment:
do context.Content.As<MediaPart> before the Is check, and check for instead of using Is<>.